opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1229,7 +1234,144 @@ namespace Opm {
+                            const Well& well_ecl = wells_ecl_[well->indexOfWell()];
+                            const auto inj_controls = Well::InjectionControls(0);
+                            const auto prod_controls = well_ecl.productionControls(summary_state);
+                            well->iterateWellEqWithSwitching(this->ebosSimulator_, dt, inj_controls, prod_controls, well_state, group_state, local_deferredLogger,  false, false);
+                            rate = -tcalc.calcModeRateFromRates(ws.surface_rates);
+                            if (ws.production_cmode == Opm::WellProducerCMode::THP)
+                                ws.thp = well_group_thp;
+                            well_group_rate += rate;
+                        }
+                    }
+                    double diff = well_group_rate - orig_target;
+                    return diff;
+                };
+
+                double well_group_rate(0.0);
+                double rate(0.0);
+                double thp(0.0);
+                double min_thp(1.0E8);
+                double max_thp(0.0);
+                for (auto& well : this->well_container_) {
+                    std::string well_name = well->name();
+                    if (group.hasWell(well_name)) {
+                        auto& ws = well_state.well(well_name);
+                        rate = -tcalc.calcModeRateFromRates(ws.surface_rates);
+                        well_group_rate += rate;
+                        thp = ws.thp;
+                        min_thp = std::min(min_thp, thp);
+                        max_thp = std::max(max_thp, thp);
+                    }
+                }
 
+                const auto upbranch = network.uptree_branch(nodeName);
+                const auto it = node_pressures_.find((*upbranch).uptree_node());
+                const double nodal_pressure = it->second;
+                double well_group_thp = nodal_pressure;
+
+                if ((reportStepIdx <= 1) || (well_group_rate > 0.9*orig_target)) {
+                    std::array<double, 2> range;
+                    if (well_group_thp_prev_ == 0) {
+                        range = {0.9*min_thp, 1.1*max_thp};
+                    }
+                    else {
+                        range =  {0.9* this->well_group_thp_prev_, 1.1*this->well_group_thp_prev_}; 
+                    }
+
+                    double low, high;
+
+                    std::optional<double> approximate_solution;
+                    const double limit = 1.0E-3;
+                    local_deferredLogger.debug("Using brute force search to bracket the common THP");
+                    const bool finding_bracket = WellBhpThpCalculator::bruteForceBracketCommonTHP(mismatch, range, low, high, approximate_solution, limit, local_deferredLogger);
+
+
+                    if (approximate_solution.has_value()) {
+                        well_group_thp = *approximate_solution;
+                        double check = mismatch(well_group_thp);
```

Comment:
function mismatch() involves a solution for the well and it is good to avoid run it again.  You can get the value from the previous call of function
`bruteForceBracketCommonTHP()` since it is calculated there already.

Reply:
The recalculation was used for debugging purposes so I removed it